instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcSpanStatusExtractor.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.grpc.v1_6;
+
+import io.grpc.Status;
+import io.grpc.StatusException;
+import io.grpc.StatusRuntimeException;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanStatusExtractor;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+final class GrpcSpanStatusExtractor implements SpanStatusExtractor<GrpcRequest, Status> {
+  @Override
+  public StatusCode extract(GrpcRequest grpcRequest, Status status, @Nullable Throwable error) {
+    if (status == null) {
+      if (error instanceof StatusRuntimeException) {
+        status = ((StatusRuntimeException) error).getStatus();
+      } else if (error instanceof StatusException) {
+        status = ((StatusException) error).getStatus();
+      }
```

Comment:
Don't these exceptions get unwrapped by `GrpcErrorCauseExtractor`?

Reply:
Not if there's no cause, which is a common usage for these too, e.g. just `throw Status.INVALID_ARGUMENT.asRuntimeException()`

Reply:
Hmm - I think it means one corner case for `Status.OK.withCause(fatalException).asRuntimeException()`, while `Status` is OK, the OTel `StatusCode` would be error for this... Do you think it could be better to pass the wrapped exception to this instead of unwrapped (only use unwrapped for `Span.recordException`)?

Reply:
Hmm, originally I thought that it'd be better to always unwrap the exception and pass the unwrapped one everywhere: `span.recordException()`, `SpanStatusExtractor` and `RequestListener#end()` once we add `Throwable` arg there. The `ErrorCauseExtractor` was meant to remove those meaningless exception wrappers like `ExecutionException`.
In the gRPC instrumentation maybe it'd be better not to unwrap `StatusException`/`StatusRuntimeException`? They seem to carry some information with them (the status) that should be available when setting span status.

Reply:
The main reason I added the cause extractor was to get the existing tests to pass :) But I guess `recordException` with the status exceptions could be considered better, yeah the exception message will have some information about the status.